torch/fx/experimental/symbolic_shapes.py

```diff
@@ -116,6 +118,23 @@ def guard_scalar(a):
     else:
         raise AssertionError(f"unrecognized scalar {a}")
 
+# inclusive both ways
+def constrain_range(a, *, min, max=sympy.oo):
+    if not isinstance(a, SymInt):
+        assert min <= a <= max
+        return
+    if isinstance(a.node.expr, sympy.Integer):
```

Comment:
any reason not to support Floats here ?

Reply:
Same deal, not supporting constraints on floats for now